packages/baseclient-components/src/component/LayerCarousel/LayerCarousel.tsx

```diff
@@ -0,0 +1,219 @@
+    const {
+      map
+    } = this.props;
+    const size = map.getSize();
+    if (!size) {
+      return 128;
+    }
+    const ratio = size[0] / size[1];
+    return Math.round(128 * ratio);
+  }
+
+  /**
+   *
+   * @param evt
+   */
+  onSlideClick(layer: any) {
+    const {
+      map
+    } = this.props;
+
+    // TODO: slow clicks will be handled as drags....
+    if (this.state.mouseDownTime > 180) {
+      return;
+    }
+
+    map.getLayers().getArray().forEach((l: any) => l.set('visible', false));
+    layer.set('visible', true);
+    this.setState({
+      renderTrigger: this.state.renderTrigger + 1
+    });
+  }
+
+  /**
+   * `onMouseDown` handler for carousel. Will be used to init a mouseDownTime
+   * state value to determine between drag and click events on carousel items
+   * (s. #onSlideClick method).
+   */
+  mouseDown() {
+    this.setState({
+      mouseDownTime: new Date().getTime()
+    });
+  }
+
+  /**
+   * `onMouseUp` handler for carousel. Will be used to compute a final value of
+   * mouseDownTime state value to determine between drag and click events on
+   * carousel items (s. #onSlideClick method).
+   */
+  mouseUp() {
+    this.setState({
+      mouseDownTime: new Date().getTime() - this.state.mouseDownTime
+    });
+  }
+
+  /**
+   * The render function
+   */
+  render() {
+    const {
+      layers
+    } = this.props;
+
+    const carouselSettings = {
+      className: 'carousel',
+      slidesToShow: Math.round(window.innerWidth / this.getWidth() - 1),
+      slidesToScroll: 1,
+      swipeToSlide: true,
+      centerMode: true,
+      infinite: true,
+      lazyLoad: true,
+      dots: false,
+      draggable: true,
+      arrows: true,
+      responsive: [
+        {
+          breakpoint: 480,
+          settings: {
+            slidesToShow: 1
+          }
+        },
+        {
+          breakpoint: 768,
+          settings: {
+            slidesToShow: 2
+          }
+        },
+        {
+          breakpoint: 1024,
+          settings: {
+            slidesToShow: 3
+          }
+        },
+        {
+          breakpoint: 1366,
+          settings: {
+            slidesToShow: 4
+          }
+        },
+        {
+          breakpoint: 1920,
+          settings: {
+            slidesToShow: 5
+          }
+        }
+      ]
+    };
+
+    const layerSlides = layers.map((layer: any) =>
+      <div
+        key={Math.random()}
+        className={layer.get('visible') ? 'carousel-slide selected' : 'carousel-slide'}
+        style={{width: this.getWidth() + 'px'}}
+        onClick={this.onSlideClick.bind(this, layer)}
+        onMouseDown={this.mouseDown.bind(this)}
+        onMouseUp={this.mouseUp.bind(this)}
+      >
+        {layer.get('name')}
+        <br/>
+        <img
+          src={layer.get('imageUrl') ? layer.get('imageUrl') : this.getLayerPreview(layer)}
+          height={128}
+        />
+      </div>
+    );
+
+    return (
+      <div className='carousel-wrapper'>
```

Comment:
Wrapping div is unneeded just add `className` directly to the `Carousel`.

Reply:
changing this totally breaks the carousel, so it will stay for now